src/Run/Setup.cs

```diff
@@ -57,6 +93,35 @@ private string ParseSettingValue(string inputValue)
             return value;
         }
 
+        private void ParseRunToolSettings(string commandSelectedByUser = null)
+        {
+            Setting tempSetting;
+            Command tempCommand;
+            foreach(var toolSetting in ToolSettings.Keys.Select(k => k.ToString()).ToArray())
+            {
+                string tempValue = null;
+                // Attempt to get run tool setting value from parameters
+                SettingParameters.TryGetValue(toolSetting, out tempValue);
+                ToolSettings[toolSetting] = tempValue;
+
+                // Attempt to get run tool setting value from command section
+                if (string.IsNullOrEmpty(ToolSettings[toolSetting]) && commandSelectedByUser != null)
+                {
+                    if (Commands.TryGetValue(commandSelectedByUser, out tempCommand))
+                    {
+                        tempCommand.DefaultValues.Settings.TryGetValue(toolSetting, out tempValue);
+                        ToolSettings[toolSetting] = tempValue;
+                    }
+                }
+                // Attempt to get run tool setting value from settings section
+                if (string.IsNullOrEmpty(ToolSettings[toolSetting]))
+                {
+                    Settings.TryGetValue(toolSetting, out tempSetting);
```

Comment:
The behavior here is to look for it as a parameter, then part of the command, then settings as a last default value.  I think this behavior meets your desire.